src/algebra/char_p/algebra.lean

```diff
@@ -0,0 +1,66 @@
+/-
+Copyright (c) 2021 Jon Eugster. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Eugster, Eric Wieser
+-/
+import algebra.char_p.basic
+import ring_theory.localization
+
+
+/-!
+# Characteristics of algebras
+
+In this file we describe the characteristic of `R`-algebras.
+
+In particular we are interested in the characteristic of the fraction field
+`fraction_ring R`.
+
+
+## Main results
+
+- `char_p_of_injective_algebra_map` If `R→+*A` is an injective algebra map
+  then `A` has the same characteristic as `R`.
+
+As an example, the `fraction_ring R` of an integral domain `R` has the same characteristic.
+
+-/
+
+
+/-- If the algebra map `R →+* A` is injective then `A` has the same characteristic as `R`. -/
+lemma char_p_of_injective_algebra_map {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
+  (h : function.injective (algebra_map R A)) (p : ℕ) [char_p R p] : char_p A p :=
+{ cast_eq_zero_iff := λx,
+  begin
+    rw ←char_p.cast_eq_zero_iff R p x,
+    change algebra_map ℕ A x = 0 ↔ algebra_map ℕ R x = 0,
+    rw is_scalar_tower.algebra_map_apply ℕ R A x,
+    refine iff.trans _ h.eq_iff,
+    rw ring_hom.map_zero,
+  end }
+
+/-- If the algebra map `R →+*A` is injective then `A` has the same characteristic as `R`. -/
```

Comment:
This doc-string is just a copy of the one above. It should say something about characteristic zero.

Reply:
This comment still applies,, @joneugster